frontend/app/vite.config.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { defineConfig } from "vite"
+import react from "@vitejs/plugin-react-swc"
+import viteTsconfigPaths from "vite-tsconfig-paths"
+import { default as checker } from "vite-plugin-checker"
+
+import path from "path"
+
+const BASE = "./"
+const HASH = process.env.OMIT_HASH_FROM_MAIN_FILES ? "" : ".[hash]"
+const BUILD_AS_FAST_AS_POSSIBLE =
```

Comment:
nit: should we name this more directly like `BUILD_WITHOUT_SOURCEMAP` or something like that? or did you envision using this elsewhere eventually?

Reply:
I guess I was thinking something general since the focus is for our build in the event we want to optimize more. It also makes it easy cause it's used in our original craco config, so less updates. Let me know if you feel it's important to change.

Reply:
nah, solid reasoning to keep it as is 👌